zulip_bots/zulip_bots/bots/link_shortener/link_shortener.py

```diff
@@ -0,0 +1,63 @@
+import re
+import requests
+
+class LinkShortenerHandler(object):
+    '''A Zulip bot that will shorten URLs ("links") in a conversation using the
+    goo.gl URL shortener.
+    '''
+
+    def usage(self):
+        return (
+            'Mention the link shortener bot in a conversation and then enter '
+            'any URLs you want to shorten in the body of the message. \n\n'
+            '`key` must be set in `link_shortener.conf`.')
+
+    def initialize(self, bot_handler):
+        self.config_info = bot_handler.get_config_info('link_shortener')
+
+    def handle_message(self, message, bot_handler):
+        REGEX_STR = (
+            '('
+            '(?:http|https):\/\/' # This allows for the HTTP or HTTPS protocol.
+            '[^"<>#%\{\}|\\^~[\]` ]+' # This allows for any character except
+                                      # for certain non-URL-safe ones.
+            ')'
+        )
+
+        content = message['content']
+        link_matches = re.findall(REGEX_STR, content)
+
+        shortened_links = [self.shorten_link(link) for link in link_matches]
+        link_pairs = [
+            (link_match + ': ' + shortened_link)
+                for link_match, shortened_link
+                in zip(link_matches, shortened_links)
+                if shortened_link != ''
+        ]
+        final_response = '\n'.join(link_pairs)
+
+        if final_response == '':
```

Comment:
Consider adding some text like: `No links found. Send 'help' to see usage instructions.`. Also, add a test for the case where we don't send any link.